python/chatting_sample.py

```diff
@@ -0,0 +1,392 @@
+
+        speech_config = speechsdk.SpeechConfig(subscription=speech_key, region=service_region)
+        audio_config = speechsdk.audio.AudioConfig(filename=filename)
+
+        speech_recognizer = speechsdk.SpeechRecognizer(speech_config=speech_config, audio_config=audio_config)
+
+        done = False
+
+        def stop_cb(evt: speechsdk.SessionEventArgs):
+            """callback that signals to stop continuous recognition upon receiving an event `evt`"""
+            nonlocal done
+            done = True
+
+        # Connect callbacks to the events fired by the speech recognizer
+        speech_recognizer.recognized.connect(lambda evt: result_text.append(evt.result.text))
+        # Stop continuous recognition on either session stopped or canceled events
+        speech_recognizer.session_stopped.connect(stop_cb)
+        speech_recognizer.canceled.connect(stop_cb)
+
+        # Start continuous speech recognition
+        speech_recognizer.start_continuous_recognition()
+        while not done:
+            time.sleep(.5)
+
+        speech_recognizer.stop_continuous_recognition()
+        text = "".join(result_text)
+        print("YOU: ", text)
+        return text
+
+    def call_gpt(send_text):
+        your_resource_name = "YourResourceName"
+        your_deployment_id = "YourDeploymentId"
+        api_version = "YourApiVersion"
+        your_api_key = "YourApiKey"
+
+        url = (
+            f"https://{your_resource_name}.openai.azure.com/openai/deployments/{your_deployment_id}/"
+            f"chat/completions?api-version={api_version}"
+        )
+        headers = {"Content-Type": "application/json", "api-key": your_api_key}
+        data = {"messages": [
+            {
+                "role": "system",
+                "content": (
+                    "You are a voice assistant, and when you answer questions,"
+                    "your response should not exceed 25 words."
+                )
+            },
+            {
+                "role": "user",
+                "content": send_text
+            }
+        ]}
+
+        text = json.loads(
+            requests.post(url=url, headers=headers, data=json.dumps(data)).content
+        )["choices"][0]["message"]["content"]
+        print("GPT: ", text)
+        return ('<speak version="1.0" xmlns="http://www.w3.org/2001/10/synthesis" xml:lang="en-US">'
+                f'<voice name="en-US-JennyNeural">{text}</voice>'
+                '</speak>')
+
+    def tts(text, output_filename):
+        file_config = speechsdk.audio.AudioOutputConfig(filename=output_filename)
+        speech_config = speechsdk.SpeechConfig(subscription=speech_key, region=service_region)
+        speech_synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config, audio_config=file_config)
+
+        result = speech_synthesizer.speak_ssml_async(text).get()
+        # Check result
+        if result.reason == speechsdk.ResultReason.SynthesizingAudioCompleted:
+            pass
```

Comment:
Here can add `print(f"Save synthesized waveform to {output_filename}")` to let the users know where the wave is saved.

Reply:
> Here can add `print(f"Save synthesized waveform to {output_filename}")` to let the users know where the wave is saved.

added